elasticgraph-graphql/lib/elastic_graph/graphql/filtering/filter_interpreter.rb

```diff
@@ -187,21 +187,32 @@ def process_any_satisfy_filter_expression_on_scalar_list(bool_node, filter, fiel
           end
         end
 
+        # We want to provide the following semantics for `any_of`:
+        #
+        # * `filter: {anyOf: []}` -> return no results
+        # * `filter: {anyOf: [{field: null}]}` -> return all results
+        # * `filter: {anyOf: [{field: null}, {field: ...}]}` -> return all results
         def process_any_of_expression(bool_node, expressions, field_path)
           return if expressions.nil? || expressions == {}
 
+          if expressions.empty?
+            # When our `expressions` array is empty, we want to match no documents. However, that's
+            # not the behavior the datastore will give us if we have an empty array in the query under
+            # `should`. To get the behavior we want, we need to pass the datastore some filter criteria
+            # that will evaluate to false for every document.
+            BooleanQuery::ALWAYS_FALSE_FILTER.merge_into(bool_node)
+            return
+          end
+
           shoulds = expressions.filter_map do |expression|
             build_bool_hash do |inner_bool_node|
               process_filter_hash(inner_bool_node, expression, field_path)
             end
           end
 
-          # When our `shoulds` array is empty, the filtering semantics we want is to match no documents.
-          # However, that's not the behavior the datastore will give us if we have an empty array in the
-          # query under `should`. To get the behavior we want, we need to pass the datastore some filter
-          # criteria that will evaluate to false for every document.
-          bool_query = shoulds.empty? ? BooleanQuery::ALWAYS_FALSE_FILTER : BooleanQuery.should(*shoulds)
-          bool_query.merge_into(bool_node)
+          return if shoulds.empty? || shoulds.size < expressions.size
```

Comment:
```suggestion
          return if shoulds.size < expressions.size
```

I could be wrong about this...but I think this can be simplified. There are two possibilities when `shoulds` is empty:

1. `expressions` was also empty. In that case, we return before we even get here, and the `should.empty?` check shouldn't be needed to handle it.
2. `expressions` is not empty. In that case, `shoulds.size` must be less than `expressions.size` (as 0 is less than every positive integer), and again the `should.empty?` shouldn't be needed.

Reply:
Yes, correct, I'll make the change.